quail/utils.py

```diff
@@ -19,6 +24,73 @@
 logger.addHandler(handler)
 
 
+def collect_literal_inputs(request):
+    literal_inputs = [
+        request.inputs[k][0].data
+        for k in request.inputs.keys()
+        if "data_type" in vars(request.inputs[k][0]).keys()
+    ]
+    return literal_inputs
+
+
+def validate_vector(vector):
+    try:
+        vect = robjects.r(vector)
+        if robjects.r["is.vector"](vect)[0]:
+            pass
+        else:
+            raise ProcessError("Invalid type passed for vector")
```

Comment:
Would you be able to explain the condition in the `if` statement? I'm not sure I follow what is happening. Furthermore, you may want to rework this statement such that if it is true, we raise the error. That way we don't have to have the extra `pass` and `else` statements.

Reply:
Makes sense. You can use the syntax `robjects.r["some.r.function"]` to use a R function without importing the library so alternatively I could write
```Python
base = get_package("base")
base.is_vector(vect)
```
which returns `[1] TRUE` so to actually access the actual value I have to index it with `[0]`